### PR TITLE
[qa] Fix error introduced into p2p-segwit.py, and prevent future similar errors

### PR DESCRIPTION
**Pull request by sdaftuar**
#11121 inadvertently broke the constructor for the `TestNode()` object in `p2p-segwit.py`, silently breaking at least one of the tests.

Although the python code was raising exceptions due to a `TestNode()` object not existing (or having the right type), mininode was masking these from anyone running the test through the test_runner (like travis), because it catches all exceptions during message delivery and just prints a log message and continues.  Such "graceful" handling of errors is almost certainly something we don't want in our test suite, so the first commit here attempts to prevent that type of failure from ever being masked.

The second commit fixes the particular bug in `p2p-segwit.py`.

### CODE REVIEW

**Comment by promag** [comment]
utACK f97ab35.

**Comment by promag** [comment]
Nit, commit a782042 could be on it's own PR.

**Review by jnewbery — COMMENTED** [review]
Testing this now. Definitely ACK to the `p2p-segwit.py` change. That's a crass bug introduced by me.

One question for @MarcoFalke inline. The other two exception catches have been in `mininode.py` since it was introduced in 6c1d1ba6fccd76e3690bac8341687c0921758e30, so as long as the tests all pass it should be fine to make these changes.

**Comment by jnewbery** [comment]
Tested ACK f97ab35fa9687fd5c110ad6cca5be5b4a5c2142d. Extended test suite passes locally.

**Comment by sdaftuar** [comment]
@marcofalke Anything else needed here?

File: test/functional/test_framework/mininode.py [hunk 1]
```diff
@@ -1773,8 +1771,10 @@ def got_data(self):
                     self.got_message(t)
                 else:
                     logger.warning("Received unknown command from %s:%d: '%s' %s" % (self.dstaddr, self.dstport, command, repr(msg)))
+                    raise ValueError("Unknown command: '%s'" % (command))
         except Exception as e:
             logger.exception('got_data:', repr(e))
+            raise
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply test/functional/test_framework/mininode.py]: @jnewbery As of faaa3c9b6546d9a64cece4ff0223f0b167feb6ff there was a `try,except,pass` in `handle_read` that ate all exceptions without notice. By adding the error print I didn't change behavior but added useful debug information.

**Comment by MarcoFalke** [comment]
@sdaftuar No. All looks fine. Thanks for the ping.

utACK f97ab35fa9687fd5c110ad6cca5be5b4a5c2142d

File: test/functional/test_framework/mininode.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply test/functional/test_framework/mininode.py]: Though, obviously they should be thrown. Going to merge this.

**Pull request merged**